CI/windows/build_cmocka.ps1

```diff
@@ -0,0 +1,77 @@
+Param(
+    [Switch]$Help = $(if (Test-Path variable:Help) { $Help }),
+    [Switch]$Quiet = $(if (Test-Path variable:Quiet) { $Quiet }),
+    [Switch]$Verbose = $(if (Test-Path variable:Verbose) { $Verbose }),
+    [Switch]$NoChoco = $(if (Test-Path variable:NoChoco) { $NoChoco }),
+    [Switch]$SkipDependencyChecks = $(if (Test-Path variable:SkipDependencyChecks) { $SkipDependencyChecks }),
+    [Switch]$Install = $(if (Test-Path variable:Install) { $Install }),
+    [String]$BuildDirectory = "build",
+    [ValidateSet("32-bit", "64-bit")]
+    [String]$BuildArch = (Get-CimInstance CIM_OperatingSystem).OSArchitecture,
+    [ValidateSet("Release", "RelWithDebInfo", "MinSizeRel", "Debug")]
+    [String]$BuildConfiguration = "RelWithDebInfo"
+)
+
+################################################################################
+# Windows cmocka native-compile build script
+################################################################################
+#
+# This script file can be included in build scripts for Windows or run directly
+#
+################################################################################
+
+function Build-Product {
+    cd "${DepsBuildDir}"
+
+    if ($Quiet) {
+        $CMAKE_OPTS = "-Wno-deprecated -Wno-dev --log-level=ERROR"
+    } else {
+        $CMAKE_OPTS = ""
+    }
+
+    Write-Step "Configure (${ARCH})..."
+    cmake -G "Visual Studio 16 2019" `
```

Comment:
I've come to prefer composing the CMake command this way:

```
   $CmakeCommand = @(
        "-S . -B `"${BuildDirectory}`"",
        "-G `"Visual Studio 16 2019`"",
        "-DCMAKE_GENERATOR_PLATFORM=`"${GeneratorPlatform}`"",
        "-DCMAKE_SYSTEM_VERSION=`"${CmakeSystemVersion}`"",
        "-DCMAKE_PREFIX_PATH=`"${CmakePrefixPath}`"",
        [OTHER_STUFF],
        "$(if (Test-Path Variable:$Quiet) { "-Wno-deprecated -Wno-dev --log-level=ERROR" })"
    )

    Invoke-Expression "cmake ${CmakeCommand}"
```

Reply:
Hmm.  I'll mark this down as "PowerShell-isms to investigate/implement later".